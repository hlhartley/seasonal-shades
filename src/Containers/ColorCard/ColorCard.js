import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { formatColorName } from '../../Helpers/colorsHelper';

export class ColorCard extends Component {
    constructor() {
        super()
    }
    render() {
        const formattedColor = formatColorName(this.props.color);
        const { hexcode } = this.props.allColors[formattedColor];

        if (Object.keys(this.props.allColors).length) {
            return(
                <NavLink to={`/${this.props.type}/${formattedColor}`}>
                    <div className='color' style={{ backgroundColor: hexcode }}>
                        <i className="far fa-heart"></i>
                        <p className='color-hexcode-text'>{this.props.color}</p>
                    </div>
                </NavLink>
            )
        } else {
            return (
                <p>BLAH</p>
            )
        }
    };
};

export const mapStateToProps = (state) => ({
    allColors: state.allColors,
})

export default connect(mapStateToProps)(ColorCard);