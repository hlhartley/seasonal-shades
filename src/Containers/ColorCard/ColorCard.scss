.color-hexcode-text {
    display: flex;
    justify-content: center;
    align-items: center;
}

a:hover{
    -webkit-transform: scale(1.15);
    -ms-transform: scale(1.15);
    transform: scale(1.15);
}
