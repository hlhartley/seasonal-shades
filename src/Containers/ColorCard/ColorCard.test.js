import React from 'react';
import { ColorCard, mapStateToProps } from './ColorCard';
import { shallow } from 'enzyme';

describe('ColorCard', () => {
    let wrapper;
    describe('ColorCard component', () => {
        const allColors = {
            522: {
                product: {},
                hexcode: '#EED8BE'
            },
            771: {
                product: {},
                hexcode: '#BD9E9B' 
            },
            'casablanca': {
                product: {},
                hexcode: '#444446'
            }
        }
        beforeEach(() => {
            wrapper = shallow(<ColorCard color={'casablanca'} type={'lipstick'} allColors={allColors} lipstickColors={[]} blushColors={[]} eyeshadowColors={[]} nail_polishColors={[]}/>)
        });

        it('should match the correct snapshot with all the data passed in', () => {
            expect(wrapper).toMatchSnapshot()
        });
    });

    describe('mapStateToProps', () => {
        it('should return an object with all colors', () => {
            const mockState = {
                allColors: {
                    522: {
                        product: {},
                        hexcode: '#EED8BE'
                    },
                    771: {
                        product: {},
                        hexcode: '#BD9E9B' 
                    },
                    'casablanca': {
                        product: {},
                        hexcode: '#444446'
                    }
                },
                lipstickColors: []
            }
            const expected = {
                allColors: {
                    522: {
                        product: {},
                        hexcode: '#EED8BE'
                    },
                    771: {
                        product: {},
                        hexcode: '#BD9E9B' 
                    },
                    'casablanca': {
                        product: {},
                        hexcode: '#444446'
                    }
                }
            }
            const mappedProps = mapStateToProps(mockState)
            expect(mappedProps).toEqual(expected)
        });
    });
});