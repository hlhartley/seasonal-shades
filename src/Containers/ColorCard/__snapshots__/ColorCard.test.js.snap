// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCard ColorCard component should match the correct snapshot with all the data passed in 1`] = `
<NavLink
  activeClassName="active"
  aria-current="page"
  to="/lipstick/casablanca"
>
  <div
    className="color"
    style={
      Object {
        "backgroundColor": "#444446",
      }
    }
  >
    <i
      className="far fa-heart"
    />
    <p
      className="color-hexcode-text"
    >
      casablanca
    </p>
  </div>
</NavLink>
`;
